### PR TITLE
chore: Update to react native web view metamask registry

### PR DESCRIPTION
**Pull request by tommasini**
Using @metamask/react-native-webview instead react-native-webview

### CODE REVIEW

**Comment by legobeat** [comment]
The package is also references by name in a direct import: https://github.com/MetaMask/react-native-animated-fox/blob/1622fc0d81585efc4637604d74b0116b266384aa/index.js#L3 - I assume this should be changed as well?

(Could) also consider adding the same dependency to `dependencies` with same range as the `peerDependencies` entry.

**Review by Cal-L — APPROVED** [review]
Lgtm!

**Review by legobeat — CHANGES_REQUESTED** [review]
https://github.com/MetaMask/react-native-animated-fox/pull/5#issuecomment-2121418420


The import needs to be updated or it will keep attempting to pull in the old module.

https://github.com/MetaMask/metamask-mobile/actions/runs/9166255130/job/25201356176#step:5:130

**Comment by tommasini** [comment]
Missed it! Thanks for flagging @legobeat

File: index.js [hunk 1]
```diff
@@ -1,32 +1,30 @@
-import React, { PureComponent } from 'react';
-import PropTypes from 'prop-types';
-import { WebView } from 'react-native-webview';
+import React, { PureComponent } from "react";
```
Comment (legobeat) [inline index.js:1]: @tommasini Think we can revert these unrelated style/formatting-changes like changing quotes out of this PR and keep it to the relevant change? Will make a difference when syncing this fork with upstream.

Comment (tommasini) [reply index.js]: Ah sorry, auto save tricked me

**Comment by tommasini** [comment]
@sethkfman just made a really good point that this shouldn't be a different repository if it's only used on mobile repo

**Comment by legobeat** [comment]
> @sethkfman just made a really good point that this shouldn't be a different repository if it's only used on mobile repo

What is the other repo this shouldn't be different from..?

**Review by legobeat — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
   "peerDependencies": {
     "react-native": "*",
     "react-native-sensors": "*",
-    "react-native-webview": "*",
+    "@metamask/react-native-webview": "*",
```
Comment (legobeat) [inline package.json:22]:
```suggestion
    "@metamask/react-native-webview": ">=14.0.1",
```

Since we know `14.0.0` to be broken.

**Comment by legobeat** [comment]
I added a restriction on the `peerDependencies` range of `@metamask/react-native-webview`: 0ae03df2df2412629846ebf30a82f7e06a5b57c1

**Pull request merged**